proselint/registry/checks/__init__.py

```diff
@@ -90,61 +94,57 @@ class CheckFlags(NamedTuple):
     ppm_threshold: int = 0
 
     @staticmethod
-    def truncate(results: list[CheckResult], limit: int) -> list[CheckResult]:
+    def truncate(
+        results: Iterator[CheckResult], limit: int
+    ) -> Iterator[CheckResult]:
         """
         Truncate a list of results to a given threshold.
 
         This also notes how many times the check flagged prior to truncation.
         """
-        if limit == 0 or (num_results := len(results)) <= limit:
+        if limit == 0:
             return results
 
-        last_result = results[limit - 1]
-        num_extras = num_results - limit
-        last_message = last_result.message + " Found {} elsewhere.".format(
-            "once" if num_extras == 1 else f"{num_extras} times"
-        )
-
-        return [
-            *results[0 : limit - 1],
-            CheckResult(
-                start_pos=last_result.start_pos,
-                end_pos=last_result.end_pos,
-                check_path=last_result.check_path,
-                message=last_message,
-                replacements=None,
+        return chain(
+            islice(results, limit - 1),
+            (
+                CheckResult(
+                    start_pos=result.start_pos,
+                    end_pos=result.end_pos,
+                    check_path=result.check_path,
+                    message=f"{result.message} Also found elsewhere.",
```

Comment:
Minor nitpick, unless it makes sense in this specific context.
```suggestion
                    message=f"{result.message} also found elsewhere.",
```

Reply:
The idea here is that check messages are supposed to end with a period anyway. Take, for instance, `archaism`: "'{}' is archaic." would become "'{}' is archaic. Also found elsewhere."

Reply:
Noted, I feel like it'd be better to do this in this function itself rather than rely on the check's messages, though that'd be a lot of work. If necessary, we could always add validation within the check constructor itself.